python/lsst/ts/observatory/control/remote_group.py

```diff
@@ -766,7 +777,14 @@ async def set_state(self, state, settings=None, components=None):
             else:
                 set_ss_tasks.append(self.get_state(comp, ignore_timeout=True))
 
-        ret_val = await asyncio.gather(*set_ss_tasks, return_exceptions=True)
+        ret_val = (
+            await asyncio.gather(*set_ss_tasks, return_exceptions=True)
+            if self._concurrent_operation
+            else [
+                (await asyncio.gather(task, return_exceptions=True))[0]
```

Comment:
Can this be simplified to `await task for task in set_ss_tasks` or is it done this way to keep an exception on one task from blowing up the whole execution?

Reply:
yes.. the idea is to execute all tasks and, if one or more fails, return an error with that information.